sycl/source/detail/program_manager/program_manager.cpp

```diff
@@ -659,15 +663,37 @@ RTDeviceBinaryImage &ProgramManager::getDeviceImage(OSModuleHandle M,
   return *Img;
 }
 
+static bool isDeviceLibRequired(DeviceLibExt Ext, uint32_t DeviceLibReqMask) {
+  static constexpr uint32_t DeviceLibAssert = 0x1;
+  static constexpr uint32_t DeviceLibCmath = 0x2;
+  static constexpr uint32_t DeviceLibCmath64 = 0x4;
+  static constexpr uint32_t DeviceLibComplex = 0x8;
+  static constexpr uint32_t DeviceLibComplex64 = 0x10;
```

Comment:
moreover, we don't really need them, and can use cl_intel_devicelib_assert,... + bit shifts to create masks

Reply:
@andykaylor , is there a mechanism, BTW, to share definitions between the SYCL runtime and llvm tools? It then could be used for few other things as well.

Reply:
I don't think we've defined a formal mechanism. In general, the LLVM libraries are designed around a layering concept, which isn't documented but can be deduced from the cmake dependencies. When new classes or definitions are added, you need to find the right layer to put them in so that they can be shared among all the things that depend on them.

For a simple constant definition like this, it doesn't need to be complicated. Just add a header file at the lowest common level. Things like class definitions are more complicated because you have to make sure there are no circular dependencies or similar problems.

Reply:
> Just add a header file at the lowest common level.

right, but what is this level between the llvm and runtime? Basically what we need here I guess is a mechanism to have a static library (or at least headers) which can be used both from llvm tools and SYCL runtime.

Reply:
Hi, @andykaylor  and @kbobrovs 
I think what you are suggesting is to put the the definition shared between llvm-tools and sycl runtime in a low level header and include it in both llvm tools and sycl runtime.
We may have different options:
1. created a new header in llvm/include/....
2. created a new header in sycl/include/CL/sycl...
Which one do you prefer or do you have other idea?
Thank you very much.

Reply:
Hi, @andykaylor  and @kbobrovs 
To share definition between llvm tools and sycl runtime, I added SYCLRTShared.h in llvm/Support and moved the definition of "enum DeviceLibExt" to this header file. This header file will be included in sycl-post-link.cpp and program_manager.hpp. Could you help me review it?
Thank you very much.